### PR TITLE
Add terminal.terminalWidth for windows & posix/unix-like

### PR DESCRIPTION
**Pull request by c-blake**
Hey...So, for most nice formatting for terminal like output, it is usually helpful to know at least the current terminal width.  This PR adds that proc to the standard lib.  It should compile/run/work on Windows, though it would be best for someone with a commonly used Windows Nim build environment to double-check that.

### CODE REVIEW

File: lib/posix/termios.nim [hunk 1]
```diff
@@ -259,3 +259,13 @@ proc tcFlow*(fd: cint; action: cint): cint {.importc: "tcflow",
 # Get process group ID for session leader for controlling terminal FD.
 
 proc tcGetSid*(fd: cint): Pid {.importc: "tcgetsid", header: "<termios.h>".}
+
+# Window size ioctl.  Should work on on any Unix that xterm has been ported to.
+var TIOCGWINSZ*{.importc, header: "<sys/ioctl.h>".}: culong
+
+type ioctl_winsize* {.importc: "struct winsize", header: "<termios.h>",
```
Comment (Araq) [inline lib/posix/termios.nim]: Make that IOctlWinsize or something to adhere to NEP-1.

File: lib/pure/terminal.nim [hunk 2]
```diff
@@ -60,6 +60,22 @@ when defined(windows):
     lpConsoleScreenBufferInfo: ptr CONSOLE_SCREEN_BUFFER_INFO): WINBOOL{.stdcall,
     dynlib: "kernel32", importc: "GetConsoleScreenBufferInfo".}
 
+  proc terminalWidth*(h: Handle): int =
```
Comment (Araq) [inline lib/pure/terminal.nim]: For consistency with the posix target, make that

proc terminalWidthIoctl*(fds: openArray[int]): int

**Comment by c-blake** [comment]
Well, if we want it to be closer between the two, it should also take an openArray of file handles like the Unix part.  Assuming you would agree, I made that change, too.  This is getting further from code I know successfully worked on Windows.  It would be good to test this on Windows if you have a second to nim c --run some file with "import terminal; echo terminalWidth()" in it and see if you get a reasonable number of columns.

**Comment by c-blake** [comment]
Another naming possibility would be to just call both procs terminalWidth (i.e., drop the Ioctl suffix) in both the windows and posix parts and just rely on overloading to select the intended one.  That might be cleaner.  I expect most users to just want terminalWidth with no arguments, but thought the file handle version worthy enough of export.

**Comment by Araq** [comment]
I like it this way and can confirm it works on Windows.

**Pull request merged**